### ISSUE TITLE
[Bug]: Jaws is not announcing any information after invoking any button

### ISSUE THREAD

**Issue by salmaatarekk** [microsoft/fluentui#24524]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
OS: Windows 11 Version 22H2 (OS build: 22622.290)
Browser: Chrome Version 104.0.5112.79 (Official Build) (64-bit)
Screen reader: Jaws
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/salmatarek23/pen/xxWNZew

## Actual Behavior
After invoking any button Jaws announce 'Enter' only and doesn't announce that the button has been clicked 

try to run JAWS on the above codepen snippet

## Expected Behavior
Jaws should announce that the button has been clicked successfully

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
Hi @salmaatarekk, thanks for testing with screen readers and logging the issue! JAWS does not announce the successful clicking of a button unless the button has a pressed/not pressed state that toggles when clicked. This is common across all screen readers -- a click on its own is not something to announce, though a change in state or change in keyboard focus would be.

If you have users complaining about the lack of feedback when performing an action, the way to resolve that would depend on the specific action taken. For example:

- A button that opens a dialog or other popup should cause keyboard focus to move into the popup. That way, screen readers will announce the change in focus as a way to notify the user that an action occurred.
- A button in an editing toolbar (e.g. the one on this github comment editor) that toggles bold/not bold formatting should have `aria-pressed="true/false"`, and the change in `aria-pressed` will be announced by screen readers.
- A button that triggers a notification message, like a file upload progress/success. In that case, the announcement should be implemented through the notification message, likely by using a progressbar and live region.

TL:DR; a standard button on its own will not and should not generate any custom screen reader notification of a click. If the click causes something to happen on the page, any screen reader feedback will be handled by the thing that happened.
